### DISCUSSION TITLE
watch() returns a different object

### DISCUSSION THREAD

**Discussion by fenchi** [react-hook-form/react-hook-form#5662] [Q&A] [answered]
Hello,

I have a problem with the watch on an object. Each time you return from watch, the result is a new object but with exactly the same data (except when there is really a change).
I need this not to change with each render because I am using it with react-table.

For now, I am forced to use JSON.stringify and React.useMemo but I find it very dirty:
```
const watchData = watch()
const data = React.useMemo(()=>watchData, [JSON.stringify(watchData)])
```

Is there another watch-like feature that doesn't return a new item every time?

Thank you

**Comment by timothyallan** [comment, marked as answer]
I ran into this as well and ended up using the following pattern:
```
  useEffect(() => {
    watch((formValues) => {
      doStuff(formValues);
    });
  }, [doStuff, watch]);
```
  Maybe you could update a `ref` in there which holds the form data, which wouldn't cause a re-render.

**Reply by fenchi** [reply]
Great, thanks for the tip. It is much less dirty.
With the useRef it doesn't cause rendering so I managed it with a useState.
It works wonderfully.
